src/libraries/System.Security.Cryptography.Pkcs/src/System/Security/Cryptography/Pkcs/CmsSigner.cs

```diff
@@ -185,7 +185,18 @@ internal void CheckCertificateValue()
             newSignerInfo.DigestAlgorithm.Algorithm = DigestAlgorithm.Value!;
             byte[] dataHash;
 
-            using (IncrementalHash hasher = IncrementalHash.CreateHash(hashAlgorithmName))
+            IncrementalHash hasher;
+
+            try
+            {
+                hasher = IncrementalHash.CreateHash(hashAlgorithmName);
+            }
+            catch (PlatformNotSupportedException ex)
+            {
+                throw new CryptographicException(SR.Format(SR.Cryptography_UnknownHashAlgorithm, hashAlgorithmName), ex);
```

Comment:
How come in some places we thrown CryptographicException with UnknownHashAlgorithm and in other places we thrown PlatformNotSupportedException?  Is this just being consistent with the exceptions that already exist in each of the relevant places?

Reply:
> Is this just being consistent with the exceptions that already exist in each of the relevant places?

Right. Especially for S.S.C.Pkcs which also supports .NET Framework. .NET Framework is going to throw CryptographicException when it encounters a SHA3 signed CMS or tries to produce one. We want the exception in .NET to match so the exception types don't differ by platform.